### PR TITLE
Fix resource leak v4l2 output

### PR DESCRIPTION
**Pull request by shoffmeister**
### Description

Fix resource leak occurring on error path in Linux v4l2-output.

### Motivation and Context

Resource leaks generally are to be avoided.

### How Has This Been Tested?

Manual code review.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — CHANGES_REQUESTED** [review]
Per our [Contributing guildelines](https://github.com/obsproject/obs-studio/blob/b1c65af2b1c2e7c3554efb0578ca92fd64c4519c/CONTRIBUTING.rst), please make these changes:
1. Use the correct commit prefix ("linux-v4l2: ").
2. Write commit message subjects in imperative mood (begin commit message subjects with a verb).

**Comment by shoffmeister** [comment]
Many thanks for the feedback; I have adjusted the commit messages as requested.

**Review by kkartaltepe — APPROVED** [review]
Looks good to me, thanks.

**Review by RytoEX — APPROVED** [review]
Commit messages look good now. Deferring functional review to @kkartaltepe , who has approved.

File: plugins/linux-v4l2/v4l2-output.c [hunk 1]
```diff
@@ -75,8 +75,7 @@ static bool loopback_module_loaded()
 		}
 	}
 
-	if (fp)
-		fclose(fp);
+	fclose(fp);
```
Comment (norihiro) [inline plugins/linux-v4l2/v4l2-output.c:78]: When `fp` is NULL, `fclose` should not be called.
```suggestion
	if (fp)
		fclose(fp);
```

**Comment by WizardCM** [comment]
Was this leak introduced in 28, or is it a longer-standing bug?

**Comment by kkartaltepe** [comment]
Looks like the FD leak was there since the original commit.

**Pull request merged**